nix/repl-containers.nix

```
@@ -0,0 +1,31 @@
+{ pkgs }:
+let
+  mkImage = pkg: cmd:
+    pkgs.dockerTools.buildLayeredImage {
+      name = "ghcr.io/cofree-coffee/${pkg.name}-repl";
+      created = "now";
+      tag = "latest";
+      contents = [
+        pkg
+      ];
+      config = {
+        Cmd = cmd;
+      };
+    };
+
+  mkPath = xs:
+    with pkgs.lib;
+    with pkgs.lib.attrsets;
+    let paths = mapAttrsToList (name: image: "ln -s ${image} $out/${name}") xs;
+    in "mkdir -p $out" + (foldl' (a: b: a + ";" + b) "" paths);
+
+  images =
+    {
+      python = mkImage pkgs.python3 [ "python" "-iq" ];
+      node = mkImage pkgs.nodejs [ "node" "-i" ];
+      ghci = mkImage pkgs.ghc [ "ghci" ];
+      mitScheme = mkImage pkgs.mitscheme [ "mit-scheme" ];
+      sbcl = mkImage pkgs.sbcl [ "sbcl" ];
+    };
```

```suggestion
  images =
    foldl' (acc {name, package, cmd}: acc // { "${name}" = mkImage package cmd }) {} supportedLanguages;
  supportedLanguages =
    [ { name = "python"; package = pkgs.python3; cmd = [ "python" "-iq" ]}
      { name = "node"; package = pkgs.nodejs; cmd = [ "node" "-i]}
      { name = "ghci"; package = pgks.ghc; cmd = [ "ghci"]}
      { name = "mitScheme"; package = pkgs.mitscheme; cmd = [ "mit-scheme ]
      { name = "sbcl"; package = pkgs.sbcl; cmd = [ "sbcl"]}
    ];
```
By more scalable do you mean something like this (like easier to add, although I am not sure this is an improvement). Or do you mean that a lot of docker containers is expensive.